stl/src/vector_algorithms.cpp

```diff
@@ -1496,5 +1497,121 @@ __declspec(noalias) size_t
     return __std_count_trivial<_Find_traits_8>(_First, _Last, _Val);
 }
 
+} // extern "C"
+
+namespace {
+    __m128i
+#ifdef _M_X64
+        __vectorcall
+#endif
+        _Bitset_to_string_1_step(const uint16_t _Val, const __m128i _Elem0_v, const __m128i _Elem1_v) {
+        const uint16_t _Valsw = _byteswap_ushort(_Val);
+        const __m128i _Vec1   = _mm_cvtsi32_si128(_Valsw);
+        const __m128i _Vec2   = _mm_unpacklo_epi8(_Vec1, _Vec1);
+        const __m128i _Vec3   = _mm_unpacklo_epi8(_Vec2, _Vec2);
+        const __m128i _Vec4   = _mm_unpacklo_epi8(_Vec3, _Vec3);

```

Comment:
If you swap the last unpacklo for shuffle_epi32, you can then save one instruction by merging the byteswap into that shuffle.